src/Fields/mapreduce_cuda.jl

```diff
@@ -150,42 +150,33 @@ function mapreduce_cuda_kernel!(
     tidx = threadIdx().x
     bidx = blockIdx().x
     fidx = blockIdx().y
+    dataview = _dataview(pdata, fidx)
```

Comment:
Why not just pass the `data` object directly to the kernel? that way, we can just use regular datalayout indexing.

Reply:
That would need a few additional changes, including moving away from linear indexing. I would prefer to do that in a different PR, if that is ok!

Reply:
but i don't see how this fixes the issue?